src/commitment.rs

```diff
@@ -4,10 +4,14 @@ use crate::polynomial;
 use crate::setup;
 use blst;
 
-pub struct Proof {}
+pub struct Proof {
+    pub qs_in_g1: blst::blst_p1,
+}
 
 pub struct Opening {
+    pub input: point::Point,
     pub value: point::Point,
+    pub fs_in_g1: blst::blst_p1,
     pub proof: Proof,
```

Comment:
theoretically, i'd like to hide the actual type for a `proof`, perhaps that is something like:

```rust
type Proof = blst::blst_p1;

pub struct Opening {
    pub value: point::Point,
    pub proof: Proof,
}
```

and leave `Proof` private?

Reply:
I'm not a fan of `type` (and it doesn't hide anything). If we want to make `Proof` opaque I think we'd need a struct

```rust
pub struct Proof {
    // private field
    proof: blst::blst_p1,
}
```

For now I've just left it as a `blst::blst_p1`.